### PR TITLE
test: use az resources for Key Vault integration tests

### PR DESCRIPTION
**Pull request by stijnmoreels**
Use a real Azure Key Vault to test our key vault scripts in the integration tests.

Relates to #174

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,6 +8,7 @@
 *.user
 *.userosscache
 *.sln.docstates
+*.local.json
```
Comment (stijnmoreels) [inline .gitignore:11]: Added this so we can test locally.

**Comment by stijnmoreels** [comment]
Have to change some test-related features so it installs the correct packages. Not directly related with this PR, but to this repo. Apparenly a new Az package was released two days ago. That's the failing cause.
Working on it...

**Comment by stijnmoreels** [comment]
> Have to change some test-related features so it installs the correct packages. Not directly related with this PR, but to this repo. Apparenly a new Az package was released two days ago. That's the failing cause.
> Working on it...

Voila, that's done. More stable now.

File: src/Arcus.Scripting.KeyVault/Scripts/Set-AzKeyVaultSecretFromFile.ps1 [hunk 2]
```diff
@@ -32,4 +33,4 @@ if ($Expires -ne $null) {
 }
 
 $version = $secret.Version
-Write-Host "Secret '$SecretName' (Version: '$version') has been created."
+Write-Host "Azure Key Vault Secret '$SecretName' (Version: '$version') has been created."
```
Comment (mbraekman) [inline src/Arcus.Scripting.KeyVault/Scripts/Set-AzKeyVaultSecretFromFile.ps1:36]: Can you update the docs as well, to reflect these updates in output?

**Review by mbraekman — COMMENTED** [review]
Seems to be incorporating everything it should:
- Use of app settings + tokenize
- Use of BeforeEach instead of repeating per test
- Clean-up of created resources (secrets in this case)
Great job!

Also thanks for the improvements on the outputs. Could you just make sure to update the documentation to reflect these changes?

File: src/Arcus.Scripting.KeyVault/Scripts/Get-AzKeyVaultAccessPolicies.ps1 [hunk 3]
```diff
@@ -4,55 +4,45 @@ param(
...
+             certificates = $keyVaultAccessPolicy.PermissionsToCertificates
+             storage = $keyVaultAccessPolicy.PermissionsToStorage
+          }
+
+          $armAccessPolicy | Add-Member -MemberType NoteProperty -Name permissions -Value $armAccessPolicyPermissions
+          $armAccessPolicies += $armAccessPolicy
+       }   
+    }    
+} else {
+    Write-Warning "Azure Key Vault '$keyVaultName' could not be found, please check if the vault has the correct name and is located in the resource group you specified"
```
[the remark this answers is not in the document]
Comment (stijnmoreels) [reply src/Arcus.Scripting.KeyVault/Scripts/Get-AzKeyVaultAccessPolicies.ps1:40]: No problem 😄 , is great.

File: src/Arcus.Scripting.KeyVault/Scripts/Set-AzKeyVaultSecretFromFile.ps1 (hunk 2 above)
Comment (stijnmoreels) [reply src/Arcus.Scripting.KeyVault/Scripts/Set-AzKeyVaultSecretFromFile.ps1:36]: Sure, thanks!

**Review by mbraekman — APPROVED** [review]

**Pull request merged**